plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/di/scope/DiagramGlobal.java

```diff
@@ -0,0 +1,37 @@
+/********************************************************************************
+ * Copyright (c) 2021 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package org.eclipse.glsp.server.di.scope;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import com.google.inject.ScopeAnnotation;
+
+/**
+ *
+ * Custom scope annotation that is used to declare bindings as diagram global. Diagram global means that one instance of
+ * this class is shared across all related injectors of the same diagram type. This is used to declare diagram type
+ * specific singletons across all diagram session injectors (i.e. child injectors) of one GLSP server (injector).
+ *
+ */
+@Target({ TYPE, METHOD })
+@Retention(RUNTIME)
+@ScopeAnnotation
+public @interface DiagramGlobal {}
```

Comment:
Should we rename that to `DiagramGlobalSingleton`?

Reply:
Or maybe `DiagramTypedSingleton`? To make it more clear that these scope provides Singletons in the context of the diagram type

Reply:
Yes sure, I'd prefer `DiagramTypeSingleton` a bit over `DiagramTypedSingleton` or `DiagramGlobalSingleton`.